ephios/plugins/questionnaires/signals.py

```diff
@@ -0,0 +1,144 @@
+    signup_formfields,
+    signup_save,
+)
+from ephios.core.signup.forms import AdditionalField, AdditionalFields, BaseSignupForm
+from ephios.core.signup.participants import AbstractParticipant, LocalUserParticipant
+from ephios.extra.permissions import PermissionField
+from ephios.plugins.questionnaires.forms import QuestionnaireForm
+from ephios.plugins.questionnaires.models import Question
+
+
+@receiver(nav_link, dispatch_uid="ephios.plugins.questionnaires.signals.nav_link")
+def add_nav_link(sender, request, **kwargs):
+    return (
+        [
+            {
+                "label": _("Questions"),
+                "url": reverse_lazy("questionnaires:question_list"),
+                "active": request.resolver_match
+                and request.resolver_match.app_name == "questionnaires",
+                "group": _("Management"),
+            }
+        ]
+        if request.user.has_perm("questionnaires.view_question")
+        else []
+    )
+
+
+@receiver(
+    register_group_permission_fields,
+    dispatch_uid="ephios.plugins.questionnaires.signals.register_group_permission_fields",
+)
+def group_permission_fields(sender, **kwargs):
+    return [
+        (
+            "manage_questions",
+            PermissionField(
+                label=_("Manage Questions"),
+                help_text=_("Enables this group to add questions that can be asked during signup."),
+                permissions=[
+                    "questionnaires.view_question",
+                    "questionnaires.add_question",
+                    "questionnaires.change_question",
+                    "questionnaires.delete_question",
+                ],
+            ),
+        )
+    ]
+
+
+@receiver(shift_forms, dispatch_uid="ephios.plugins.questionnaires.signals.shift_forms")
+def question_selection_form(sender, shift, request, **kwargs):
+    return [QuestionnaireForm(request.POST or None, shift=shift)]
+
+
+@receiver(signup_formfields, dispatch_uid="ephios.plugins.questionnaires.signals.signup_formfields")
+def provide_signup_formfields(
+    sender,
+    shift: Shift,
+    participant: AbstractParticipant,
+    participation: AbstractParticipation,
+    signup_choice,
+    **kwargs
+):
+    if signup_choice == BaseSignupForm.CHOICE_DECLINE:
+        return None
+
+    formfields = [
+        question.get_signup_formfield(participant, participation, signup_choice)
+        for question in Question.objects.filter(pk__in=shift.questionnaire)
+    ]
+
+    can_save_answers = isinstance(participant, LocalUserParticipant)
+    formfields.append(
+        AdditionalField(
+            "save_answers",
+            forms.BooleanField,
+            {
+                "label": _("Save answers to my user profile for future sign-ups"),
+                "initial": can_save_answers,
+                "disabled": not can_save_answers,
+                "required": False,
+                "widget": forms.HiddenInput if not can_save_answers else forms.CheckboxInput,
```

Comment:
is there a reason why you still add the field as hidden if it can't be saved anyways?

Reply:
I'm reading the value in <https://github.com/lukasrad02/ephios/blob/00988c7c22f49657f9f4b0ef5899566f8aacfe5a/ephios/plugins/questionnaires/signals.py#L135>. By always adding the field, I don't have to handle any special cases when saving the form.

Reply:
you could also just check whether `save_answers in cleaned_data` to make it simpler, but its totally fine this way as well